src/Component/OpenApi3/Tests/fixtures/discriminator-in-response/expected/Endpoint/GetObjectsNoMapping.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Jane\Component\OpenApi3\Tests\Expected\Endpoint;
+
+class GetObjectsNoMapping extends \Jane\Component\OpenApi3\Tests\Expected\Runtime\Client\BaseEndpoint implements \Jane\Component\OpenApi3\Tests\Expected\Runtime\Client\Endpoint
+{
+    use \Jane\Component\OpenApi3\Tests\Expected\Runtime\Client\EndpointTrait;
+    public function getMethod() : string
+    {
+        return 'GET';
+    }
+    public function getUri() : string
+    {
+        return '/objects-no-mapping';
+    }
+    public function getBody(\Symfony\Component\Serializer\SerializerInterface $serializer, $streamFactory = null) : array
+    {
+        return array(array(), null);
+    }
+    public function getExtraHeaders() : array
+    {
+        return array('Accept' => array('application/json'));
+    }
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \Jane\Component\OpenApi3\Tests\Expected\Exception\GetObjectsNoMappingBadRequestException
+     * @throws \Jane\Component\OpenApi3\Tests\Expected\Exception\GetObjectsNoMappingNotFoundException
+     *
+     * @return null|\Jane\Component\OpenApi3\Tests\Expected\Model\ObjectOne|\Jane\Component\OpenApi3\Tests\Expected\Model\ObjectTwo
+     */
+    protected function transformResponseBody(string $body, int $status, \Symfony\Component\Serializer\SerializerInterface $serializer, ?string $contentType = null)
+    {
+        if (is_null($contentType) === false && (200 === $status && mb_strpos($contentType, 'application/json') !== false)) {
+            $decoded = $serializer->decode($body, 'json');
+            if (isset($decoded['type']) && $decoded['type'] === 'ObjectOne') {
+                return $serializer->denormalize($decoded, 'Jane\\Component\\OpenApi3\\Tests\\Expected\\Model\\ObjectOne');
+            }
+            if (isset($decoded['type']) && $decoded['type'] === 'ObjectTwo') {
+                return $serializer->denormalize($decoded, 'Jane\\Component\\OpenApi3\\Tests\\Expected\\Model\\ObjectTwo');
+            }
```

Comment:
I'm not a big fan of handling polymorphism during response denormalization.
I do think it would be better to handle it in generated normalizers, it will allow us much more flexibility.
With this implementation, with multiple `oneOf` combined I don't think it will work :thinking:

Reply:
Hi, finally got some time to look into this, so I agree regarding polymorphism but normalizer are fetched based on object being normalized and that information I can only get if I decode data first, right ?

If we want to move this logic to normalizer maybe solution is to introduce some sort of  "Descriminator normalizer" which will decide based on data which normalizer ( based on object from `oneOf` directive) to invoke further ?

Also regarding having multiple `oneOf` combined when handling polymorphism within response, can you point me on swagger documentation describing this ?

Reply:
Hey @nestcode, thanks for looking this PR !

Actually I do think we should have a "parent" normalizer that can regroup all possibles classes and a Normalizer (as of today) for each classes. This is probably what you meant by "Discriminator normalizer" but I prefer to be more descriptive on that point.

About having "multiple" `oneOf` combined with something like this :point_down: it won't work since you have a `oneOf` on the response *and* in an object. But if you start implementing this with normalizers, it would always work in theory.
```
"responses": {
    "200": {
        "description": "success",
        "content": {
            "application/json": {
                "schema": {
                    "oneOf": [
                        {
                            "$ref": "#/components/schemas/ObjectOne"
                        },
                        {
                            "$ref": "#/components/schemas/ObjectTwo"
                        }
                    ],
                    "discriminator": {
                        "propertyName": "type"
                    }
                }
            }
        }
    }
}
```

```
"components": {
    "schemas": {
        "ObjectOne": {
            "type": "object",
            "required": ["type"],
            "properties": {
                "type": {
                    "type": "string"
                },
                "name": {
                    "type": "string"
                },
                "pet": {
                    "oneOf": [
                        {
                            "$ref": "#/components/schemas/Cat"
                        },
                        {
                            "$ref": "#/components/schemas/Dog"
                        }
                    ],
                    "discriminator": {
                        "propertyName": "type"
                    }
                }
            }
        },
        "ObjectTwo": {
            "type": "object",
            "required": ["type"],
            "properties": {
                "type": {
                    "type": "string"
                },
                "name": {
                    "type": "string"
                }
            }
        }
    }
}
```

Reply:
Exactly , "parent" normalizer is word I was looking for :) 

Oh by multiple `oneOf` you mean in object and in response, ok I will start working on this and also include tests for this case, thanks for example.

Reply:
is this comment related only to contents of tests? or it also requires some tweaks in `GetTransformResponseBodyTrait`?

I'm asking because also need support for multiple responses with one response code and going to create another pr to get this feature 

@Korbeil

Reply:
Hi @Korbeil 

I found this feature very demanded for me also and wanted to help to finish this PR

But have some doubts or question on your idea about parent (de)normalizer. What would be the name for such normalizers? In cases with [discriminator](https://github.com/janephp/janephp/blob/next/src/Component/OpenApi3/Tests/fixtures/discriminator/expected/Normalizer/PetNormalizer.php) there are Cat and Dog derived from Pet  and it is clearly pointed in [Swagger specification](https://github.com/janephp/janephp/blob/next/src/Component/OpenApi3/Tests/fixtures/discriminator/swagger.json#L33) that Parent Normalizer should be for Pet 

But here we do not have some specific Parent Model which can be used as a class for normalizer like  `return $serializer->denormalize($decoded, 'Jane\\Component\\OpenApi3\\Tests\\Expected\\Model\\ParentModel');`. We just have `ObjectOne` and `ObjectTwo`. 

So I think the solution originally implemented in this PR is good enough. It resolves OneOf contents right in place where it appears (except your notice that `oneOfs` inside objects won't be resolved, but here I think it also should be resolved in normalizer which works with object itself) i.e in example below it must be done in ObjectOneNormalizer 

```json
        "ObjectOne": {
            "type": "object",
            "required": ["type"],
            "properties": {
                "type": {
                    "type": "string"
                },
                "name": {
                    "type": "string"
                },
                "pet": {
                    "oneOf": [
                        {
                            "$ref": "#/components/schemas/Cat"
                        },
                        {
                            "$ref": "#/components/schemas/Dog"
                        }
                    ],
                    "discriminator": {
                        "propertyName": "type"
                    }
                }
            }

```

Reply:
For such case you could name that "parent" normalizer by the property name ?